### ISSUE TITLE
Proper way of handling enumerations.

### ISSUE THREAD

**Issue by alexanderweber** [wix/accord#61]
We are having some external Proto definitions, containing type enumeration, which need to be validated on through Accord. I just want to make sure using the most convenient method for doing that.

So lets say there is a case class `Foo`:

```Scala
case class Foo(bar: String, `type`: ExternalLib.Type = ExternalLib.Type.Default)
```

It refers to a Proto type defined somewhere externally. The way we implemented the validator so far is following:

```Scala
implicit val fooValidator: Validator[Foo] = {
    val defaultValidator: Validator[Foo] = validator[Foo] { foo =>
      foo.bar is notEmpty
      // Some other validation needed in a Foo object having type Default
      // But also some common validations shared among all Foos
    }

    val emptyValidator: Validator[Foo] = validator[Foo] { foo =>
      foo.bar is empty
      // Some other validation needed in a Foo object having type Empty
      // But also some common validations shared among all Foos
    }

    new Validator[Foo] {
      override def apply(foo: Foo): Result = foo.`type` match {
        case ExternalLib.Type.Default => validate(foo)(defaultValidator)
        case ExternalLib.Type.Empty => validate(foo)(emptyValidator)
        case _ => Failure(Set(RuleViolation(c.`type`, "unknown", None)))
      }
  }
```

Despite the fact that the actual ugliness of the validation might lie in the model structure, I would like to ask, if there is some prettier way of validating that particular model here. Generally I would ask for a best practise validation of objects containing enumerated properties.

**Comment by holograph** [comment]
I think this might well be a special case of #58 (and specifically #59). Your workaround is a reasonable one, but this definitely needs some work on the macro transformation. I'll try and get on that in the next few weeks.

**Comment by holograph** [comment]
Closed as duplicate of #58 and #59.
